koku/api/report/queries.py

```diff
@@ -85,36 +88,35 @@ def __init__(self, query_parameters, url_data,
             units_key   (String): the key defining the units
             kwargs    (Dict): A dictionary for internal query alteration based on path
         """
-        self.query_parameters = query_parameters
-        self.url_data = url_data
-        self.tenant = tenant
-        self.aggregate_key = aggregate_key
         self._accept_type = None
-        self._filter = None
-        self._group_by = None
         self._annotations = None
         self._count = None
-        self._delta = self.query_parameters.get('delta')
-        self._limit = self.get_query_param_data('filter', 'limit')
-        self._order = self.get_order()
-        self.operation = self.query_parameters.get('operation', OPERATION_SUM)
-        self.units_key = units_key
+        self._filter = None
+        self._group_by = None
+        self.end_datetime = None
         self.resolution = None
-        self.time_scope_value = None
-        self.time_scope_units = None
         self.start_datetime = None
-        self.end_datetime = None
         self.time_interval = []
+        self.time_scope_units = None
+        self.time_scope_value = None
+
+        self.aggregate_key = aggregate_key
+        self.query_parameters = query_parameters
+        self.tenant = tenant
+        self.url_data = url_data
+
+        self.operation = self.query_parameters.get('operation', OPERATION_SUM)
+        self._delta = self.query_parameters.get('delta')
+        self._limit = self.get_query_param_data('filter', 'limit')
         self._get_timeframe()
+        self.units_key = units_key
 
         if kwargs:
-            elements = ['accept_type', 'annotations', 'delta',
+            elements = ['accept_type', 'annotations', 'count', 'delta',
```

Comment:
Adding `count` here and using `setattr` overrides the property on the class. I had it as a property, and left the view using `resource_id` to allow for the path where the original `AWSCostEntryLineItem` table is used. In that case the referenced column is `resource_id` The property setter is supposed to override the value to `resource_count` when the summary table is used. If we are certain we never need to use the original resource_id column, we can do this, but then the property should be deleted. If we are uncertain let's not use setattr here and instead let the property handle it.

Reply:
I'll look a bit more closely at this.

This is where leaving code comments about what's going on and why is super-helpful to call out this kind of subtle behavior. So much of the ReportQueryHandler is un-documented, which makes working on it a giant pain.

Though, I also think it definitely needs reworking to make these behaviors more well-defined and obvious.